insights/parsers/ros_config.py

```diff
@@ -0,0 +1,140 @@
+"""
+RosConfig - file ``/var/lib/pcp/config/pmlogger/config.ros``
+======================================================
+This class provides parsing for the files:
+    ``/var/lib/pcp/config/pmlogger/config.ros``
+
+Sample input data is in the format::
+
+log mandatory on default {
+    mem.util.used
+    mem.physmem
+    kernel.all.cpu.user
+    kernel.all.cpu.sys
+    kernel.all.cpu.nice
+    kernel.all.cpu.steal
+    kernel.all.cpu.idle
+    kernel.all.cpu.wait.total
+    disk.all.total
+    mem.util.cached
+    mem.util.bufmem
+    mem.util.free
+}
+[access]
+disallow .* : all;
+disallow :* : all;
+allow local:* : enquire;
+"""
+
+from insights import Parser
+from insights.parsr import EOF, EOL, Char, Literal, Many, OneLineComment, Opt, QuotedString, String, WSChar
+import string
+
+
+# https://man7.org/linux/man-pages/man1/pmlogger.1.html#CONFIGURATION_FILE_SYNTAX
+
+WS = Many(WSChar | EOL | OneLineComment("#"))
+
+Log = WS >> Literal("log") << WS
+MandatoryOn = WS >> Literal("mandatory on") << WS
+MandatoryOff = WS >> Literal("mandatory off") << WS
+MandatoryMaybe = WS >> Literal("mandatory maybe") << WS
+AdvisoryOn = WS >> Literal("advisory on") << WS
+AdvisoryOff = WS >> Literal("advisory off") << WS
```

Comment:
As ROS specific config file uses ` mandatory on default` operation so do you really need to handle rest of the operations?

Reply:
Sure @kgaikwad, we can make the parser more compact based on specific ROS related configs.